### PR TITLE
ignore locked statics in taskDefend

### LINKED ISSUE #1013: "CBA_fnc_taskDefend" does not ignore locked statics.
**Arma 3 Version:** `1.84` (stable)
**CBA Version:** `3.9.0` (stable)

**Mods:**
```
- CBA_A3
- ACE3
- CUP Weapons/Vehicles/Units
- ACRE2
```

**Description:**
AI groups given the taskDefend function will mount statics that are fully locked (not just locked for players). 

**Steps to reproduce:**
- Place AI group.
- Give AI group CBA_fnc_taskDefend.
- Place static near them.
- Lock the static.

**Where did the issue occur?**
- Dedicated

### PR DESCRIPTION
**Pull request by Cuel**
**When merged this pull request will:**
- AI will not consider locked statics in taskDefend

fix #1013

### CODE REVIEW

File: addons/ai/fnc_taskDefend.sqf [hunk 1]
```diff
@@ -58,7 +58,7 @@ private _statics = _position nearObjects ["StaticWeapon", _radius];
 private _buildings = _position nearObjects ["Building", _radius];
 
 // Filter out occupied statics
-_statics = _statics select {(_x emptyPositions "Gunner") > 0};
+_statics = _statics select {locked _x != 2 && {(_x emptyPositions "Gunner") > 0}};
```
Comment (commy2) [inline addons/ai/fnc_taskDefend.sqf]: 0 - Unlocked; 1 - Default; 2 - Locked; 3 - Locked for player; -1 - vehicle is null.

Comment (Cuel) [reply addons/ai/fnc_taskDefend.sqf]: You want to also check for -1?

**Comment by commy2** [comment]
Same for https://github.com/CBATeam/CBA_A3/blob/master/addons/ai/fnc_waypointGarrison.sqf and https://github.com/CBATeam/CBA_A3/blob/master/addons/modules/fnc_moduleDefend.sqf
?

File: addons/ai/fnc_taskDefend.sqf (hunk 1 above)
Comment (commy2) [reply addons/ai/fnc_taskDefend.sqf]: No, but this is really messy.

**Comment by Cuel** [comment]
Fixed the first one but moduleDefend uses CBA_fnc_taskDefend which is fixed in first commit

**Comment by Cuel** [comment]
On that note it might also be worth check for `assignedGunner` in these aswell, if taskDefend runs for two groups they should try to assign to the same static even if it's assigned but not mounted?

**Comment by commy2** [comment]
Don't think that amount of micromanaging is necessary. It is possible for another AI or player to enter the vehicle before the assigned AI can reach for example, and that would still be not covered by that check.
At that point, it is the responsibility of the mission maker.

**Comment by commy2** [comment]
@Cuel ready to merge?

**Comment by Cuel** [comment]
Yap

**Pull request merged**